tests/unit/DBSQLClient.test.js

```
@@ -91,23 +106,49 @@ describe('DBSQLClient.openSession', () => {
         return true;
       },
     };
-    return client.openSession().then((session) => {
+
+    case1: {
+      const session = await client.openSession({ initialCatalog: 'catalog' });
       expect(session).instanceOf(DBSQLSession);
-    });
+    }
+
+    case2: {
+      const session = await client.openSession({ initialSchema: 'schema' });
+      expect(session).instanceOf(DBSQLSession);
+    }
+
+    case3: {
+      const session = await client.openSession({ initialCatalog: 'catalog', initialSchema: 'schema' });
+      expect(session).instanceOf(DBSQLSession);
+    }
   });
 
-  it('should throw an exception when the connection is lost', (done) => {
+  it('should throw an exception when not connected', async () => {
+    const client = new DBSQLClient();
+    client.connection = null;
+
+    try {
+      await client.openSession();
+      expect.fail('It should throw an error');
+    } catch (error) {
+      expect(error.message).to.be.eq('DBSQLClient: connection is lost');
+    }
+  });
+
+  it('should throw an exception when the connection is lost', async () => {
```

This is almost same as the previous one - just different invalid connection status. If we have the third case, would you add another "complete" test case?
You could consolidate into a a base test method on given connection objects and the test cases themselves just pass in different invalid connections and expect the same behaviors.

it('should throw an exception when the connection is in an invalid state', async () => {
  for (<null_connection>, <disconnected_connection>) {
    verifyConnectionError(connection)
  }
}

verifyConnectionError () => {
}

---

Those test cases look similar, but they check a different branches in the code - when connection object wasn't created at all, and when it's available but not connected. For now both branches have similar behavior (throw an error), but I'd probably not merge these cases together. Especially considering that both are quite small, simple, and easy to read